firebase_admin/mlkit.py

```diff
@@ -69,7 +76,107 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
-    #TODO(ifielker): define the Model properties etc
+    @property
+    def name(self):
+        return self._data['name']
+
+    @property
+    def display_name(self):
+        return self._data['displayName']
+
+    #TODO(ifielker): define the rest of the Model properties etc
+
+
+class ListModelsPage(object):
+    """Represents a page of models in a firebase project.
+
+    Provides methods for traversing the models included in this page, as well as
+    retrieving subsequent pages of models. The iterator returned by
+    ``iterate_all()`` can be used to iterate through all the models in the
+    Firebase project starting from this page.
+    """
+    def __init__(self, list_models_func, list_filter, page_size, page_token):
+        self._list_models_func = list_models_func
+        self._list_filter = list_filter
+        self._page_size = page_size
+        self._page_token = page_token
+        self._list_response = list_models_func(list_filter, page_size, page_token)
+
+    @property
+    def models(self):
+        """A list of Models from this page."""
+        return [Model(model) for model in self._list_response.get('models', [])]
+
+    @property
+    def list_filter(self):
+        """The filter string used to filter the models."""
+        return self._list_filter
+
+    @property
+    def next_page_token(self):
+        return self._list_response.get('nextPageToken', '')
+
+    @property
+    def has_next_page(self):
+        """A boolean indicating whether more pages are available."""
+        return bool(self.next_page_token)
+
+    def get_next_page(self):
+        """Retrieves the next page of models if available.
+
+        Returns:
+            ListModelsPage: Next page of models, or None if this is the last page.
+        """
+        if self.has_next_page:
+            return ListModelsPage(
+                self._list_models_func,
+                self._list_filter,
+                self._page_size,
+                self.next_page_token)
+        return None
+
+    def iterate_all(self):
+        """Retrieves an iterator for Models.
+
+        Returned iterator will iterate through all the models in the Firebase
+        project starting from this page. The iterator will never buffer more than
+        one page of models in memory at a time.
+
+        Returns:
+            iterator: An iterator of Model instances.
+        """
+        return ModelIterator(self)
+
+
+class ModelIterator(object):
```

Comment:
Rename to `_ModelIterator`. This doesn't need to be in the public API.

Reply:
done